### ISSUE TITLE
IFile/IListItem - CheckIfUserHasPermissionsAsync Throws ->System.NullReferenceException:

### ISSUE THREAD

**Issue by denbg9fe** [pnp/pnpcore#1097]
## Category

- [x] Bug

## Describe the bug

CheckIfUserHasPermissionAsync for IListItem (coming from IFile) throws "System.NullReferenceException: 'Object reference not set to an instance of an object.'"

It's happening in 
ListItem.cs file line  -> 1679 - > Method "BuildGetUserEffectivePermissionsApiCall"

Seems that the Parent.Parent is not a list, it's a Web object.
`var parentList = Parent.Parent as List;`
Then it fails during string formatting because parentList is null

## Steps to reproduce

```
var file = await clientContext.Web.GetFileByServerRelativeUrlAsync(relativeUrl, 
            f => f.ListItemAllFields.QueryProperties(p => p.ParentList), f => f.Parent);
var item = file.ListItemAllFields;

//BUG in PnP.Core - Workaround - Uncomment if you want that it works
//item.Parent.Parent = item.ParentList;

return await item.CheckIfUserHasPermissionsAsync(spUser.UserPrincipalName, permission);
```

## Expected behavior

To not do the workaround

```
var file = await clientContext.Web.GetFileByServerRelativeUrlAsync(relativeUrl, 
            f => f.ListItemAllFields.QueryProperties(p => p.ParentList), f => f.Parent);
var item = file.ListItemAllFields;

return await item.CheckIfUserHasPermissionsAsync(spUser.UserPrincipalName, permission);
```

## Environment details (*development & target environment*)

- **SDK version**: 1.8.0
- **OS**: Windows 10
- **SDK used in**: Console App & ASP.Net Web app
- **Framework**: net6.0
- **Tooling**: VS 2022

**Comment by jansenbe** [comment]
Thanks for using PnP Core SDK and reporting this @denbg9fe , I'll have a look.

**Comment by jansenbe** [comment]
@denbg9fe : already fixed this one :-) Fix will appear in next nightly (version 1.8.100 or higher). Closing this issue now.
